code/Controllers/CMSMain.php

```diff
@@ -1084,14 +1084,19 @@ public function Breadcrumbs($unlinked = false)
         $ancestors = $record->getAncestors();
         $ancestors = new ArrayList(array_reverse($ancestors->toArray() ?? []));
         $ancestors->push($record);
+
+        $params = $this->getRequest()->getVars();
         /** @var SiteTree $ancestor */
         foreach ($ancestors as $ancestor) {
             $items->push(new ArrayData([
                 'Title' => $ancestor->getMenuTitle(),
                 'Link' => ($unlinked)
                     ? false
-                    : $ancestor->CMSEditLink()
-            ]));
+                    : Controller::join_links(
+                        $ancestor->CMSEditLink(),
+                        '?' . http_build_query($params ?? [])
+                    )
+                ]));
```

Comment:
I don't think we want this - we probably only want this on the "back" button in the breadcrumb rather than each link.

If we _do_ want this to apply to each link, we should do this from the `GridFieldDetailForm_ItemRequest::Breadcrumbs()` method. This already has a list of all of the breadcrumbs (`$items = $this->popupController->Breadcrumbs($unlinked);`) which we can loop through like this:

```php
foreach ($items as $item) {
    if ($item->Link) {
        $item->Link = $this->gridField->addOtherGridfieldStateToURL($item->Link);
    }
}
```
That way we only have this in the context of a gridfield, instead of arbitrarily adding all params to every breadcrumb.